src/Precompiler.sol

```diff
@@ -48,56 +79,80 @@ contract Precompiler {
         }
     }
 
-    function run_sha256(uint256 gas_target) private {
+    function run_sha256(uint256 gas_target, bool use_long) private {
         uint256 start_gas = gasleft();
         uint256 gas_used = 0;
 
+        uint256 count = 0;
         while (gas_used < gas_target) {
-            sha256(abi.encodePacked(gas_used));
+            count += 1;
+            if (use_long) {
+                sha256(abi.encodePacked(hashLongString()));
+            } else {
+                sha256(abi.encodePacked(gas_used));
+            }
             gas_used = start_gas - gasleft();
         }
+        console.log("SHA256 count: %d", count);
     }
 
-    function run_ripemd160(uint256 gas_target) private {
+    function run_ripemd160(uint256 gas_target, bool use_long) private {
         uint256 start_gas = gasleft();
         uint256 gas_used = 0;
 
+        uint256 count = 0;
         while (gas_used < gas_target) {
-            ripemd160(abi.encodePacked(gas_used));
+            count += 1;
+            if (use_long) {
+                ripemd160(abi.encodePacked(hashLongString()));
+            } else {
+                ripemd160(abi.encodePacked(gas_used));
+            }
             gas_used = start_gas - gasleft();
         }
+        console.log("RIPEMD160 count: %d", count);
     }
 
-    function run_identity(uint256 gas_target) private {
+    function run_identity(uint256 gas_target, bool use_long) private {
         uint256 start_gas = gasleft();
         uint256 gas_used = 0;
 
+        uint256 count = 0;
         while (gas_used < gas_target) {
-            address(4).staticcall(abi.encode(gas_used));
+            count += 1;
+            if (use_long) {
+                address(4).staticcall(abi.encode(hashLongString()));
+            } else {
+                address(4).staticcall(abi.encode(gas_used));
+            }
             gas_used = start_gas - gasleft();
         }
+        console.log("Identity count: %d", count);
     }
 
-    function run_modexp(uint256 gas_target) private {
+    function run_modexp(uint256 gas_target, bool use_long) private {
         uint256 start_gas = gasleft();
         uint256 gas_used = 0;
 
         bytes memory base = "8";
         bytes memory exponent = "9";
+        uint256 count = 0;
         while (gas_used < gas_target) {
-            bytes memory modulus = abi.encodePacked(gas_used);
-            address(5).staticcall(
-                abi.encodePacked(
-                    base.length,
-                    exponent.length,
-                    modulus.length,
-                    base,
-                    exponent,
-                    modulus
-                )
-            );
+            count += 1;
+            if (use_long) {
+                bytes memory modulus = abi.encodePacked(hashLongString());
```

Comment:
If `hashLongString` were to return `bytes` instead of a `string`, then you would not need to call `abi.encodePacked` on this again.